packages/react-native/scripts/cocoapods/new_architecture.rb

```diff
@@ -49,18 +49,11 @@ def self.modify_flags_for_new_architecture(installer, is_new_arch_enabled)
         unless is_new_arch_enabled
             return
         end
-        ndebug_flag = " -DNDEBUG"
         # Add RCT_NEW_ARCH_ENABLED to Target pods xcconfig
         installer.aggregate_targets.each do |aggregate_target|
             aggregate_target.xcconfigs.each do |config_name, config_file|
                 config_file.attributes['OTHER_CPLUSPLUSFLAGS'] = @@new_arch_cpp_flags
 
-                if config_name == "Release"
-                    config_file.attributes['OTHER_CPLUSPLUSFLAGS'] = config_file.attributes['OTHER_CPLUSPLUSFLAGS'] + ndebug_flag
-                    other_cflags = config_file.attributes['OTHER_CFLAGS'] != nil ? config_file.attributes['OTHER_CFLAGS'] : "$(inherited)"
-                    config_file.attributes['OTHER_CFLAGS'] = other_cflags + ndebug_flag
-                end
```

Comment:
The problem with the test is that, by removing these lines, the function `modify_flags_for_new_architecture` does not alter the `OTHER_CFLAGS` attribute anymore.

So, now, when we execute `modify_flags_for_new_architecture`, nothing access the `attribute["OTHER_CFLAGS"]` which remains `nil`. Instead, the test that is failing expect to find the `"$(inherithed)"` string.